### PR TITLE
fix (kubernetes-model-generator) : Update Kubernetes Model to v1.25.0 (#4347)

### LINKED ISSUE #4327: Fix Dependabot reported alerts in go modules
### Describe the bug

Dependabot has reported a few alerts on our internal go module tooling

https://github.com/fabric8io/kubernetes-client/security/dependabot

### Fabric8 Kubernetes Client version

SNAPSHOT

### Steps to reproduce

n/a

### Expected behavior

No issues should be reported

### Runtime

other (please specify in additional context)

### Kubernetes API Server version

other (please specify in additional context)

### Environment

other (please specify in additional context)

### Fabric8 Kubernetes Client Logs

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

Fix #4327 #4347

+ Update Kubernetes Model to v1.25.0
   - `clusterName` removed from [ObjectMeta](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L260)
   - Add a test for CustomResourceDefinition validation expression language
   - Add resources in `k8s.io/api/flowcontrol/v1beta2` API (they were present earlier too, just realized it a bit late) 
   - Add `autoscaling/v2` HorizontalPodAutoscaler model and DSL 
+ Force `containerd/containerd`, `docker/distribution` and
  `opencontainers/image-spec` to not use vulnerable versions (#4327)
+ Kubernetes Model Generator script currently converted all byte, integer lists to string (since there is no byte reflect.type in GoLang , it's alias to `uint8`). This seems to be incorrectly generating PodFailurePolicyOnExitCodesRequirement.values field as String. Add a switch case in `javaType()` to treat `reflect.Uint8` to byte. Limit `javaTypeArrayList()` to only convert byte array/ `reflect.Uint8` array to string.
+ Update E2E test matrix suite to add Kubernetes 1.25.0

Signed-off-by: Rohan Kumar <rohaan@redhat.com>

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [X] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) entry regarding this change
 - [X] I have implemented unit tests to cover my changes
 - [X] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by yelowstar** [comment]
Please add support for apiVersion autoscaling/v2 since now is using Fabric8 Kubernetes Client 5.11.2 which doesn't seem to have it.

File: kubernetes-itests/src/test/java/io/fabric8/kubernetes/CronJobIT.java [hunk 1]
```diff
@@ -29,6 +30,7 @@
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 @LoadKubernetesManifests("/cronjob-it.yml")
+@RequireK8sSupport(CronJob.class)
```
Comment (manusa) [inline kubernetes-itests/src/test/java/io/fabric8/kubernetes/CronJobIT.java:33]: Follow-up update this test to use CronJob v1 instead

Comment (manusa) [reply kubernetes-itests/src/test/java/io/fabric8/kubernetes/CronJobIT.java:33]: Issue created: https://github.com/fabric8io/kubernetes-client/issues/4434

File: kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodEvictIT.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+import java.util.concurrent.TimeUnit;
+
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@LoadKubernetesManifests("/pod-evict-it.yml")
+@RequireK8sSupport(PodDisruptionBudget.class)
```
Comment (manusa) [inline kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodEvictIT.java:38]: Is there any reason why this new test is based on v1beta1 of PodDisruptionBudget?

**Review by manusa — APPROVED** [review]

File: kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodEvictIT.java (hunk 2 above)
Comment (rohanKanojia) [reply kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodEvictIT.java:38]: This isn't a new test. This was earlier part of PodIT . This test was breaking on Kubernetes 1.25 hence I moved it to separate class with a `RequireK8sSupport` condition

Comment (manusa) [reply kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodEvictIT.java:38]: Follow-up: update this test to use PodDisruptionBudget v1 instead

Comment (manusa) [reply kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodEvictIT.java:38]: Issue created: https://github.com/fabric8io/kubernetes-client/issues/4436

**Pull request merged**